primary/src/metrics.rs

```diff
@@ -56,6 +56,12 @@ pub struct PrimaryMetrics {
     pub batches_received: IntCounterVec,
     /// Latency to perform a garbage collection in core module
     pub gc_core_latency: HistogramVec,
+    /// The current Narwhal round
+    pub current_round: IntGaugeVec,
+    /// Latency to perform a garbage collection in header_waiter
+    pub gc_header_waiter_latency: HistogramVec,
+    /// Number of elements in pending list of header_waiter
+    pub pending_elements_header_waiter: IntGaugeVec,
```

Comment:
Same with the above comment, since we are basically getting a "snapshot" of the pending elements on the list - which doesn't necessarily correlate with the previous report instance (we don't always expect this to go up) I've used the gauge